lib/features/mailbox/data/repository/mailbox_repository_impl.dart

```diff
@@ -152,4 +171,59 @@ class MailboxRepositoryImpl extends MailboxRepository {
   Future<bool> renameMailbox(AccountId accountId, RenameMailboxRequest request) {
     return mapDataSource[DataSourceType.network]!.renameMailbox(accountId, request);
   }
+
+  @override
+  Future<bool> markAsMailboxRead(AccountId accountId, MailboxId mailboxId, MailboxName mailboxName) async {
```

Comment:
this is a heavy task in some case, how about this task block the main isolate for a while?

Reply:
I created a new ticket #616 to improve that.